internal/beater/beater.go

```diff
@@ -612,15 +620,19 @@ func (s *Runner) newFinalBatchProcessor(
 		v.OnInt(indexer.Stats().Added)
 	})
 	monitoring.Default.Remove("output")
-	monitoring.NewFunc(monitoring.Default, "output.elasticsearch.bulk_requests", func(_ monitoring.Mode, v monitoring.Visitor) {
+	monitoring.NewFunc(monitoring.Default, "output.elasticsearch.indexers", func(_ monitoring.Mode, v monitoring.Visitor) {
```

Comment:
Not quite what I had in mind. This will produce:

- `output.elasticsearch.indexers.active` (currently active number of indexers)
- `output.elasticsearch.indexers.available` (sounds like available indexers, is actually available bulk requests)
- `output.elasticsearch.indexers.created` (counter of indexers that have been created)
- `output.elasticsearch.indexers.destroyed` (counter of indexers that have been destroyed)
- `output.elasticsearch.indexers.requests_completed` (counter of bulk requests that have been completed)

So I see a couple of issues:
 - the existing "active bulk requests" metric (`output.elasticsearch.bulk_requests.active`) has been removed but probably shouldn't have been
 - `output.elasticsearch.indexers.available` doesn't describe what it really is

Active bulk requests counts the `*bulkIndexers` that have been acquired from, but not yet released back to, the "available" channel.

I suggest reinstating all the existing metrics, and just introducing new `output.elasticsearch.indexers.*` metrics. Here's what I would expect to see at the end:

- `output.elasticsearch.indexers.active` (currently active number of indexers)
- `output.elasticsearch.indexers.created` (counter of indexers that have been created)
- `output.elasticsearch.indexers.destroyed` (counter of indexers that have been destroyed)
- `output.elasticsearch.bulk_requests.active` (currently active number of bulk requests)
- `output.elasticsearch.bulk_requests.available` (currently available number of bulk requests)
- `output.elasticsearch.bulk_requests.completed` (counter of bulk requests that have been completed)

Reply:
Sounds much better, I may not have thought the metric name change through.